iset.mm

```diff
@@ -95229,6 +94988,217 @@ gives an ordered pair whose first element is the index (translated
+        OZUVGVEZUVLUVFUVBYQSUVBUVOTARYQUUFFYSVOUVBYPGWDUSUVFUVNUVJUVGUVFBCDEFGH
+        IJUUFAEWESUVEKWFADGSUVELWFAGHULUVEMWFABUNZYPSCUNZGSVDUVPUVQIWGGSUVENWHO
+        AUVEWIPWJWLWMWNUVMUVKUVGYKTZUVJUVGYJYKUAWOZQWOZWPWQWRAUUTUVEUVKUVIVFUVA
+        RYPUUFYJJWSVNXBUVIUULUVBVJUUKUUFFWTXAXCXMUVCUVRUVFYJYKUVBUVSUVTXDXEXFXG
+        XHXIXJWFUUIUUQUBUUMGUUFUUMTZUUHUUPQUWAUUGUUOUUCUUFUUMYKXKXLXNXOXPUUCQUB
+        UUCUBXQXRUSAUUEYTAYNYPYJABCDEFGHIJKLMNOPXSXTYAYLUUCQYJYKYHYBYCYDYEUAQYH
+        YFYG $.
+    $}
+
+    ${
+      $d C x y z $.  $d F x y z $.  $d R x y $.  $d S x y z $.  $d T x y z $.
+      $d ph x y z $.
+      frecuzrdgtclt.3 $e |- ( ph -> P = ran R ) $.
+      $( The recursive definition generator on upper integers is a function.
+         (Contributed by Jim Kingdon, 22-Apr-2022.) $)
+      frecuzrdgtclt $p |- ( ph -> P : ( ZZ>= ` C ) --> S ) $=
+        ( vz wceq cz c1 cuz cfv wfn cxp wss wf wfun cdm crn cv caddc cmpt cfrec
+        oveq1 cbvmptv ax-mp frecuzrdgfun funeqd mpbird dmeqd frecuzrdgdom eqtrd
+        co freceq1 df-fn sylanbrc com frecuzrdgrclt frn syl eqsstrd dff2 ) AFEU
+        AUBZUCZFVMHUDZUEVMHFUFAFUGZFUHZVMRVNAVPGUIZUGABCDEGHIJQSQUJZTUKVCZULZEU
+        MZKLMNOWABSBUJZTUKVCZULZRWBWEEUMRQBSVTWDVSWCTUKUNUOEWAWEVDUPZUQAFVRPURU
+        SAVQVRUHVMAFVRPUTABCDEGHIJWBKLMNOWFVAVBFVMVEVFAFVRVOPAVGVOGUFVRVOUEABCD
+        EGHIJKLMNOVHVGVOGVIVJVKVMHFVLVF $.
+    $}
+
+    ${
+      $d C x y $.  $d F x y $.  $d R x y $.  $d S x y $.  $d T x y $.
+      $d ph x y $.
+      frecuzrdg0t.ran $e |- ( ph -> P = ran R ) $.
+      $( Initial value of a recursive definition generator on upper integers.
+         (Contributed by Jim Kingdon, 28-Apr-2022.) $)
+      frecuzrdg0t $p |- ( ph -> ( P ` C ) = A ) $=
+        ( wcel cfv c0 com wfun cop wceq cuz wf frecuzrdgtclt ffun syl crn cv c1
+        caddc co cmpt2 cfrec fveq1i cvv syl2anc frec0g syl5eq wfn frecuzrdgrclt
+        cz opexg cxp ffn peano1 fnfvelrn sylancl eqeltrrd eleqtrrd funopfv sylc
+        ) AFUAZEDUBZFQEFRDUCAEUDRZHFUEVNABCDEFGHIJKLMNOPUFVPHFUGUHAVOGUIZFASGRZ
+        VOVQAVRSBCVPIBUJZUKULUMVSCUJJUMUBUNZVOUOZRZVOSGWAOUPAVOUQQZWBVOUCAEVCQD
+        HQWCKLEDVCHVDURVOVTUQUSUHUTAGTVAZSTQVRVQQATVPHVEZGUEWDABCDEGHIJKLMNOVBT
+        WEGVFUHVGTSGVHVIVJPVKEDFVLVM $.
+    $}
+
+    ${
+      $d B x y z $.  $d C x y z $.  $d F x y z $.  $d G x y $.  $d R x y $.
+      $d S x y z $.  $d T x y z $.  $d ph x y z $.
+      frecuzrdgsuct.g $e |- G = frec ( ( x e. ZZ |-> ( x + 1 ) ) , C ) $.
+      frecuzrdgsuct.ran $e |- ( ph -> P = ran R ) $.
+      $( Successor value of a recursive definition generator on upper integers.
+         See comment in ~ frec2uz0d for the description of ` G ` as the mapping
+         from ` _om ` to ` ( ZZ>= `` C ) ` .  (Contributed by Jim Kingdon,
+         29-Apr-2022.) $)
+      frecuzrdgsuct $p |- ( ( ph /\ B e. ( ZZ>= ` C ) ) ->
+          ( P ` ( B + 1 ) ) = ( B F ( P ` B ) ) ) $=
```

Comment:
This theorem does not depend on G!? Why is frecuzrdgsuct.g needed?

Reply:
Pretty much every step in the second half of this proof has a `G` in it, so it is for readability of the proof as described above.